File: Skeleton Java/java/src/main/java/de/thm/mni/compilerbau/absyn/TypeExpression.java
package de.thm.mni.compilerbau.absyn;

import de.thm.mni.compilerbau.table.TypeEntry;
import de.thm.mni.compilerbau.types.Type;

/**
 * This class is the abstract superclass of all type expressions in SPL.
 * <p>
 * A type expression is either a {@link NamedTypeExpression} or an {@link ArrayTypeExpression}.
 * They behave like a formula representing a concrete semantic {@link Type} which has to be calculated
 * during phase 4.
 */
sealed public abstract class TypeExpression extends Node permits ArrayTypeExpression, NamedTypeExpression {
    public Type typeName = null;
    public TypeExpression(Position position) {
        super(position);
    }
}

File: Skeleton Kotlin/kotlin/src/main/kotlin/de/thm/mni/compilerbau/absyn/TypeExpression.kt
package de.thm.mni.compilerbau.absyn

/**
 * This class is the abstract superclass of all type expressions in SPL.
 *
 * A type expression is either a [NamedTypeExpression] or an [ArrayTypeExpression].
 * They behave like a formula representing a concrete semantic [Type] which has to be calculated
 * during phase 4.
 */
sealed class TypeExpression(position: Position) : Node(position)
